lib/active_type/nested_attributes/nests_one_association.rb

```diff
@@ -19,6 +19,7 @@ def assign_attributes(parent, attributes)
         if id = attributes.delete(:id)
           assigned_child ||= fetch_child(parent, id)
           if assigned_child
+            id = id.to_i if assigned_child.id.is_a?(Fixnum)
```

Comment:
Thanks for the PR!

I agree this does not feel quite right, but I'm not opposed to merging it. We should use `Integer` though, otherwise Ruby 2.4 will complain.

I wonder if we could do something like
```
id_was = assigned_child.id
assigned_child.id = id
raise AssignmentError, "child record '#{@target_name}' did not match id '#{id}'" if id_was != assigned_child.id
...
```
I suppose this should cast the value properly before the comparison.

Reply:
Yes, the assignment casts the value. I ended up with:

```ruby
assigned_child.id = id
if assigned_child.id == assigned_child.id_was
  assigned_child.attributes = attributes
else
  raise AssignmentError, "child record '#{@target_name}' did not match id '#{id}'"
end
```

I also did the rebase.

Thanks!